crates/parser/src/format/sanitize/datetime.rs

```diff
@@ -0,0 +1,88 @@
+use crate::{ParseConfig, Output, format::ParseResult, ParseError};
+use time::macros::format_description;
+use serde_json::Value;
+
+struct DatetimeSanitizer {
+    from: Output,
+    default_offset: time::UtcOffset,
+}
+
+// Here we are trying to parse non-RFC3339 dates
+fn datetime_to_rfc3339(val: &mut Value, default_offset: time::UtcOffset) {
+    match val {
+        Value::String(s) => {
+            let primitive_format = format_description!(
```

Comment:
nit: I think this could use some comments explaining why we're using 2 formats and the purpose of each one.